app/services/files/create_service.rb

```diff
@@ -9,9 +9,14 @@ def commit
     def validate
       super
 
-      file_name = File.basename(@file_path)
+      if @file_path =~ Gitlab::Regex.directory_traversal_regex
```

Comment:
@stanhu I don't think we need this check if gitlab_git already does the right thing to keep us inside the repository.